### PR TITLE
Complete rework of Instructor Group details page

### PR DESCRIPTION
**Pull request by stopfstedt**
fixes https://github.com/ilios/frontend/issues/6807

### CODE REVIEW

**Review by dartajax — APPROVED** [review]
You talkin' to me? I like it - approved.

**Review by jrjohnson — CHANGES_REQUESTED** [review]
This is a great change. May be better done in a followup, but the data loading parts can be improved by pre-loading all the data in the instructor group route in the same way we do for [learner-group](https://github.com/ilios/frontend/blob/master/app/routes/learner-group.js) so all the data is pre-loaded in a single fetch and then you don't need to check for `isFulfilled` or have data loading spinners. Since most users are coming from the list you can probably add loading all related courses to [the existing data loader](https://github.com/ilios/frontend/blob/master/app/services/data-loader.js#L73) and then simple re-use that in the instructor group route.

**Comment by stopfstedt** [comment]
> This is a great change. May be better done in a followup, but the data loading parts can be improved by pre-loading all the data in the instructor group route in the same way we do for [learner-group](https://github.com/ilios/frontend/blob/master/app/routes/learner-group.js) so all the data is pre-loaded in a single fetch and then you don't need to check for `isFulfilled` or have data loading spinners. Since most users are coming from the list you can probably add loading all related courses to [the existing data loader](https://github.com/ilios/frontend/blob/master/app/services/data-loader.js#L73) and then simple re-use that in the instructor group route.

handled :heavy_check_mark:

File: app/services/data-loader.js [hunk 1]
```diff
@@ -70,6 +72,19 @@ export default class DataLoaderService extends CommonDataLoaderService {
 
     return this.#loadedLearnerGroup.get(learnerGroupId);
   }
+  async loadInstructorGroup(instructorGroupId) {
+    if (!this.#loadedInstructorGroup.has(instructorGroupId)) {
+      this.#loadedInstructorGroup.set(
+        instructorGroupId,
+        this.store.findRecord('instructor-group', instructorGroupId, {
+          reload: true,
+          include: 'users',
```
Comment (jrjohnson) [inline app/services/data-loader.js:81]: Should also include all the things to load the courses view. `offerings.session.course,ilmSessions.course, maybe others I'm not thinking of`.

**Review by jrjohnson — CHANGES_REQUESTED** [review]

File: app/services/data-loader.js (hunk 1 above)
Comment (stopfstedt) [reply app/services/data-loader.js]: added. the users and courses associated with the given instructor groups are all the extra data points we will need here.

**Review by jrjohnson — APPROVED** [review]

**Pull request merged**